_ctoApps/caesar/src/caesar/caesar-python.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Script for the local Python version of Caesar.
+ * 
+ * Notes:
+ * 1) 'caesar' variable is defined in 'caesar.js'
+ * 2) input starting with - (hyphens) is not allowed
+ *    as it's interpreted as a new console parameter
+ *    (only the case for this Python implementation)
+ * 3) Linebreaks in input also lead to Python errors
+ */
+
+
+// initialize Python editor
+let editor = CodeMirror.fromTextArea(
+    document.getElementsByClassName("CodeMirror")[0], {
+        mode: "python",
+        lineNumbers: true,
+        theme: "eclipse",
+        lineWrapping: true,
+        viewportMargin: Infinity,
+        scrollbarStyle: "native",
+        tabSize: 4,
+        indentUnit: 4,
+        smartIndent: true,
+        extraKeys: {
+            Tab: function (cm) {
+                cm.replaceSelection("    ", "end")
+            },
+        },
+    }
+)
+editor.setSize("100%", "100%")
+$("#caesar-python-local-theme-selector-caption").text(editor.options.theme)
+
+// wait until Python editor is loaded
+languagePluginLoader.then(() => {
+    $("#caesar-python-local-load-animation").remove()
+    $("#caesar-python-local-textarea-wrapper").removeClass("d-none")
+    $("#caesar-python-local-command-alert").removeClass("d-none")
+    $("#caesar-python-local-ide-options").removeClass("d-none")
+    updateLocalPythonCommandInGUI() // init command element
+    editor.refresh() // needed for editor to get visible
+})
+
+const getLocalPythonCommandLineArguments = () => {
+
+    // if message or alphabet contain ' -> escape it '"'"'
+    let message = caesar.input.value.replace(/'/g, "'\"'\"'")
+    let alphabet = caesar.alphabet.alphabet.replace(/'/g, "'\"'\"'")
+
+    let args = [
+        "caesar.py",
+        ((!caesar.encrypt.value) ? "--encrypt" : "--decrypt"), // value is negated
+        "--key", caesar.key.value,
+        "--message", message,
+        "--alphabet", alphabet
+    ]
```

Comment:
Maybe it would be a good idea to make sure that the arguments are interpreted as strings by escaping them with `"`, so for example you would have
```js
let args = [
         "caesar.py",
         ((!caesar.encrypt.value) ? "--encrypt" : "--decrypt"), // value is negated
         "--key", `"${caesar.key.value}"`,
         "--message", `"${message}"`,
         "--alphabet", `"${alphabet}"`
 ]
```

I might be wrong as I have no idea how you're parsing the arguments, but that might solve the hyphen problem.

Reply:
Yes it was similar to this before, but now they're added here 

https://github.com/ct-online/cto/blob/0e130b938509e70edd5c2535f040b07f7ca066bf/_ctoApps/caesar/src/caesar/caesar-python.js#L102-L104

as Pyodide needs the quotation marks but the GUI representation of the command does not.

Reply:
I've also tried adding the double quotation marks like you had suggested. It resulted in hyphens being accepted (as they're not the first characters in the input message then), but it also altered the output. Passing hyphens btw works when I execute the command on my local terminal, so I assume this has something to do with Pyodide. I'm still trying to find a solution ...

Reply:
I've also tried adding the double quotation marks like you had suggested. It resulted in hyphens being accepted (as they're not the first characters in the input message then), but it also altered the output. Passing hyphens btw works when I execute the command on my local terminal, so I assume this has something to do with Pyodide.

Reply:
Very strange... unfortunately I don't think I'll be able to help you with this since I never worked with Pyodide... Another option would be to encode arguments to Base64 and le python decode them but that would add some extra lines which would be weird I think...

Reply:
The following works with Pyodide, but not on my local terminal. Now I'll figure out a way to make it work there too.

```js
let args = [
    "caesar.py",
    ((!caesar.encrypt.value) ? "--encrypt" : "--decrypt"), // value is negated
    `--key=${caesar.key.value}`,
    `--message=${message}`,
    `--alphabet=${alphabet}`
]
```

Reply:
For the linebreak, maybe we should treat them as spaces... it's not ideal but I can't think of another way.